common/util/src/main/java/org/eclipse/dataspaceconnector/common/configuration/ConfigurationFunctions.java

```diff
@@ -33,4 +36,18 @@ public static String propOrEnv(String key, String defaultValue) {
         return value != null ? value : defaultValue;
     }
 
+    /**
+     * Utility method to find an unallocated port. Note that there is a race condition,
+     * the port might be allocated by the time it is used.
+     *
+     * @return a server port.
+     */
+    public static int findUnallocatedServerPort() {
```

Comment:
Can you move this method under testFixtures since it should only be accessed by tests?

Reply:
There's also a `getFreePort` method in the `TestUtils` class in testFixtures that can be replaced by this, since it handles conflicts.